### PR TITLE
drivers: kscan: increase init priority

### PR DESCRIPTION
**Pull request by gmarull**
Some kscan drivers, e.g. ft5336 will not work with Zephyr default
settings due to wrong init priority order (e.g. I2C init order is 60 and
kscan is 40). kscan priority order has been increased and some custom
priorities removed since they are no longer needed.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
We lose some configurability with this change. It would be better to just increase the default defined in `drivers/kscan/Kconfig`

File: drivers/kscan/Kconfig [hunk 1]
```diff
@@ -18,10 +18,4 @@ module = KSCAN
 module-str = kscan
 source "subsys/logging/Kconfig.template.log_config"
 
-config KSCAN_INIT_PRIORITY
-	int "Keyboard scan driver init priority"
-	default 40
-	help
-	  Keyboard scan device driver initialization priority.
-
```
Comment (avisconti) [inline drivers/kscan/Kconfig:26]: Why removing completely KSCAN_INIT_PRIORITY and not, for example, change its default to APPLICATION_INIT_PRIORITY? In this way it is possible to configure it if required by some application

Comment (gmarull) [reply drivers/kscan/Kconfig:26]: changed value and removed some custom defaults that are no longer needed.

**Review by MaureenHelm — APPROVED** [review]

File: drivers/kscan/Kconfig [hunk 2]
```diff
@@ -20,7 +20,7 @@ source "subsys/logging/Kconfig.template.log_config"
 
 config KSCAN_INIT_PRIORITY
 	int "Keyboard scan driver init priority"
-	default 40
+	default 90
```
Comment (VenkatKotakonda) [inline drivers/kscan/Kconfig:23]: Kscan in independent of other buses like i2c, how changing priority impacts the init?

**Review by avisconti — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**